### ISSUE TITLE
Feature Request: Support python-style negative axis indexing.

### ISSUE THREAD

**Issue by caisq** [PAIR-code/deeplearnjs#477]
For example, in tensorflow, `tf.reduce_mean(x, axis=-1)` means average over the last dimension. deeplearn.js currently doesn't support negative axis values. It would be nice to have that support.

Functions that may also benefit from this indexing are: sum, max, min.

**Comment by caisq** [comment]
@dsmilkov @michael-terry

**Comment by caisq** [comment]
cc @michael-terry
